src/abilities/Golden-Wyrm.js

```diff
@@ -233,78 +233,139 @@ export default (G) => {
+			_targetTeam: Team.ally,
+
+			_maxTransferAmount: 50,
 
-			// 	require() :
 			require: function () {
 				if (!this.testRequirements()) {
 					return false;
 				}
 
-				let map = G.grid.getHexMap(
-					this.creature.x - 2,
-					this.creature.y - 2,
-					0,
-					false,
-					matrices.frontnback2hex,
-				);
-				// At least one target
+				/* Golden Wyrm needs to be left with at least 1 hp after the ability. We
+				can't use ability.costs because that is a fixed value, whereas this ability
+				has a dynamic health cost. */
+				if (this.creature.health < this._maxTransferAmount + 1) {
+					this.message = 'Not enough health';
+					return false;
+				}
+
 				if (
-					!this.atLeastOneTarget(map, {
+					!this.atLeastOneTarget(this._getHexes(), {
 						team: this._targetTeam,
+						optTest: this._confirmTarget,
 					})
 				) {
 					return false;
 				}
+
 				return true;
 			},
 
-			// 	query() :
 			query: function () {
-				let ability = this;
-				let wyrm = this.creature;
+				const ability = this;
+				const wyrm = this.creature;
 
 				G.grid.queryCreature({
 					fnOnConfirm: function () {
 						ability.animation(...arguments);
 					},
+					optTest: this._confirmTarget,
 					team: this._targetTeam,
 					id: wyrm.id,
 					flipped: wyrm.flipped,
-					hexes: G.grid.getHexMap(wyrm.x - 2, wyrm.y - 2, 0, false, matrices.frontnback2hex),
+					hexes: this._getHexes(),
 				});
 			},
 
-			//	activate() :
 			activate: function (target) {
-				let ability = this;
-				ability.end();
+				this.end();
 
-				let damage = new Damage(
-					ability.creature, // Attacker
-					ability.damages, // Damage Type
-					1, // Area
-					[], // Effects
-					G,
+				// The health transferred is the creature's missing life, capped to 50.
+				const transferAmount = Math.min(
+					target.stats.health - target.health,
+					this._maxTransferAmount,
 				);
-				target.takeDamage(damage);
 
-				//remove frogger bonus if its found
-				ability.creature.effects.forEach(function (item) {
-					if (item.name == 'Offense++') {
-						item.deleteEffect();
-					}
-				});
+				target.heal(transferAmount, false, false);
+
+				/* Damage Golden Wyrm using `.heal()` instead of `.takeDamage()` to apply 
+				raw damage that bypasses dodge, shielded, etc and does not trigger further 
+				effects. */
+				this.creature.heal(-transferAmount, false, false);
+
+				// Rather than individual loss/gain health logs, show a single custom log.
+				this.game.log(
+					`%CreatureName${this.creature.id}% transfers ${transferAmount} health to %CreatureName${target.id}%`,
+				);
+
+				if (this.isUpgraded()) {
+					const regrowthBuffEffect = new Effect(
+						this.title,
+						this.creature,
+						this.creature,
+						'',
+						{
+							alterations: {
+								regrowth: 10,
+							},
+						},
+						G,
+					);
+
+					this.creature.addEffect(
+						regrowthBuffEffect,
+						`%CreatureName${this.creature.id}% gains 10 regrowth points`,
```

Comment:
It'd be great for all alterations to automatically log "x gains x" like drops, but that's a larger change for another PR.